dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs

```diff
@@ -1134,13 +1144,25 @@ private void LogUsage(ChatTokenUsage usage)
 
     private ToolCallingConfig GetToolCallingConfiguration(Kernel? kernel, OpenAIPromptExecutionSettings executionSettings, int requestIndex)
     {
+        // If both behaviors are specified, we can't handle that.
+        if (executionSettings.FunctionChoiceBehavior is not null && executionSettings.ToolCallBehavior is not null)
+        {
+            throw new ArgumentException($"{nameof(executionSettings.ToolCallBehavior)} and {nameof(executionSettings.FunctionChoiceBehavior)} cannot be used together.");
```

Comment:
Should this validation be done in the `executionSettings` itself?

As soon you set any of both you get an `InvalidOperationException`.

```suggestion
            throw new ArgumentException($"{nameof(executionSettings.ToolCallBehavior)} and {nameof(executionSettings.FunctionChoiceBehavior)} cannot be used together.");
```

Reply:
To move the validation to the execution settings, it must be done in two places: the `OpenAIPromptExecutionSettings.ToolCallBehavior` property setter and the `PromptExecutionSettings.FunctionChoiceBehavior` property setter. The `OpenAIPromptExecutionSettings.ToolCallBehavior` can be easily implemented because it's easy to access the base property from a specific class/package. However, the other check is impossible because we would have to reference a specific class/package from a base one.

Reply:
Discussed offline and agreed to keep it as is taking into account that it's a temporary solution.